storage/src/main/java/tech/pegasys/teku/storage/store/StoreTransaction.java

```diff
@@ -92,10 +93,15 @@ public void putBlockAndState(
       final SignedBeaconBlock block,
       final BeaconState state,
       final BlockCheckpoints blockCheckpoints,
-      final Optional<List<BlobSidecar>> maybeBlobSidecars) {
+      final List<BlobSidecar> blobSidecars,
+      final Optional<UInt64> maybeEarliestBlobSidecarSlot) {
     blockData.put(block.getRoot(), new TransactionBlockData(block, state, blockCheckpoints));
-    maybeBlobSidecars.ifPresent(
-        blobSidecars -> this.blobSidecars.put(block.getSlotAndBlockRoot(), blobSidecars));
+    if (!blobSidecars.isEmpty()) {
+      this.blobSidecars.put(block.getSlotAndBlockRoot(), blobSidecars);
+    }
+    if (maybeEarliestBlobSidecarTransactionSlot.isEmpty()) {
```

Comment:
shouldn't be `maybeEarliestBlobSidecarSlot` from method param?
If I am right, why no tests are failing?

Reply:
No
We don't set it again if it's already set, that's an idea. 
We set it only once.
We could override it later only when pruning.
I think it's a correct logic as we don't want to move it to the right on every batch

Reply:
Probably we could check if new value is smaller and change it only this way.
Also we could consider moving earliestBlobSidecar to the separate method but I dislike it as I want to force anyone saving blobSidecars to provide it. I think it's safer